ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AdminLogsRepairCache.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.api.query.TransactionLogBloomCacher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class AdminLogsRepairCache implements JsonRpcMethod {
+  private static final String STATUS = "Status";
+
+  private final BlockchainQueries blockchainQueries;
+
+  public AdminLogsRepairCache(final BlockchainQueries blockchainQueries) {
+    this.blockchainQueries = blockchainQueries;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ADMIN_LOGS_REPAIR_CACHE.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    final Optional<Long> blockNumber = requestContext.getOptionalParameter(0, Long.class);
+
+    blockNumber.ifPresent(
+        bn ->
+            blockchainQueries
+                .getBlockchain()
+                .getBlockByNumber(bn)
+                .orElseThrow(() -> new IllegalStateException("Block not found")));
+
+    final TransactionLogBloomCacher transactionLogBloomCacher =
+        blockchainQueries.getTransactionLogBloomCacher().orElseThrow();
+
+    transactionLogBloomCacher.ensurePreviousSegmentsArePresent(
+        blockNumber.orElse(blockchainQueries.headBlockNumber()), true);
```

Comment:
Is it possible that this won't actually force the repair of that segment? My quick reading makes it seem like we wouldn't do it if we had the right bloom file size:

https://github.com/hyperledger/besu/blob/e301b49061e1b443b2da32ae83b7075632f84518/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/TransactionLogBloomCacher.java#L220

What if the size is right but the user wants to force the recalculation because of corruption?

Reply:
The idea of the end point was to re-do if missing or the wrong size. What you are asking for is additional logic to let the user do it whenever they want. Not sure we want to add that ability to this end point.

Reply:
This has been updated to always force regeneration when called from the end point.